variables.tf

```diff
@@ -1,33 +1,53 @@
 # ------------------------------------------------------------------------------
-# REQUIRED PARAMETERS
+# OPTIONAL PARAMETERS
 #
-# You must provide a value for each of these parameters.
+# These parameters have reasonable defaults.
 # ------------------------------------------------------------------------------
 
-variable "subnet_id" {
+variable "aws_region" {
+  description = "The AWS region to deploy into (e.g. us-east-1)."
+  default     = "us-east-1"
   type        = string
-  description = "The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0)."
 }
 
-# ------------------------------------------------------------------------------
-# OPTIONAL PARAMETERS
-#
-# These parameters have reasonable defaults.
-# ------------------------------------------------------------------------------
-variable "ami_owner_account_id" {
+variable "cloudwatch_log_group_name" {
+  default     = "/ssm/session-logs"
+  description = "The name of the log group into which session logs are to be uploaded."
   type        = string
-  description = "The ID of the AWS account that owns the Example AMI, or \"self\" if the AMI is owned by the same account as the provisioner."
-  default     = "self"
 }
 
-variable "aws_availability_zone" {
+variable "cloudwatch_log_group_retention" {
+  default     = 365
+  description = "The number of days that SSM session logs will be retained in CloudWatch."
+  type        = number
+}
+
+variable "linux_shell_profile" {
+  description = "The shell setup to run when connecting to a Linux instance.  Note that this string cannot be empty."
```

Comment:
If this cannot be empty I would like validation added to this variable.

Reply:
Please see commit 103d933.